DANS-KNAW/dd-dataverse-cli
# DD-1586 Added truncate-notifications with commandline input parsing

Fixes DD-1586

# Description of changes
Implement  `dataverse truncate-notifications {--user <uid>|--all-users } <number-of-records-to-keep>`


# How to test
Deploy the role on a fresh lifesciences dev box:
```
start-preprovisioned-box.py -s dev_lifesciences
deploy.py -m dd-dataverse-cli dev_lifesciences
```
Ssh into the box and run some commands and check the result.
Example:
```
[vagrant@lifesciences ~]$ dataverse truncate-notifications --user 123 100
123: OK. Deleted 0 record(s) for user with id 123
[vagrant@lifesciences ~]$ dataverse truncate-notifications --all-users 20
Number of users found for notification truncation: 0
[vagrant@lifesciences ~]$ dataverse truncate-notifications --user 3 2
3: OK. Deleted 1 record(s) for user with id 3
[vagrant@lifesciences ~]$ dataverse truncate-notifications --all-users 2
Number of users found for notification truncation: 2
1: OK. Deleted 3 record(s) for user with id 1
8: OK. Deleted 1 record(s) for user with id 8
```

# Related PRs
(Add links)
* https://github.com/DANS-KNAW/dans-core-systems/pull/62

# Notify
@DANS-KNAW/core-systems
